examples/ui-demo/src/components/ui/dialog.tsx

```diff
@@ -0,0 +1,34 @@
+"use client";
+
+import * as React from "react";
+import * as DialogPrimitive from "@radix-ui/react-dialog";
+import { cn } from "@/lib/utils";
+
+const Dialog = DialogPrimitive.Root;
+
+const DialogTrigger = DialogPrimitive.Trigger;
+
+const DialogOverlay = DialogPrimitive.Overlay;
+
+const DialogPortal = DialogPrimitive.Portal;
+
+const DialogContent = React.forwardRef<
+  React.ElementRef<typeof DialogPrimitive.Content>,
+  React.ComponentPropsWithoutRef<typeof DialogPrimitive.Content>
+>(({ className, ...props }, ref) => (
+  <DialogPrimitive.Content
+    asChild
+    className={cn(
```

Comment:
oh I see why we have `cn` defined separately, it's because of the `twMerge` function.